cloud/grpc.go

```diff
@@ -56,19 +78,23 @@ func (c *client) StreamAuthInterceptor() grpc.StreamClientInterceptor {
 		newStreamer, err := streamer(ctx, desc, cc, method, opts...)
 		if err != nil {
 			s, ok := status.FromError(err)
-			if ok && s.Code() == codes.Unauthenticated {
-				err = c.Authenticate(ctx)
-				if err != nil {
-					return nil, errors.Wrap(err, "failed re-authenticating")
-				}
-				md, ok := metadata.FromOutgoingContext(ctx)
-				if !ok {
-					return nil, errors.New("could not parse metadata")
+			if ok {
+				err = status.Errorf(s.Code(), fmt.Sprintf("%s {reqID: %s}", s.Err(), getReqID(ctx)))
```

Comment:
Might be nice to deduplicate this code and the above.

Reply:
Yeah it was getting a bit messy as it grew, I've refactored it quite a bit now if you don't mind taking another look